diff --git a/node_modules/react-native-admob-native-ads/android/build.gradle b/node_modules/react-native-admob-native-ads/android/build.gradle
index 1cb369c..d1109b0 100755
--- a/node_modules/react-native-admob-native-ads/android/build.gradle
+++ b/node_modules/react-native-admob-native-ads/android/build.gradle
@@ -53,7 +53,7 @@ repositories {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    implementation "com.google.android.gms:play-services-ads:${safeExtGet('googlePlayServicesAdsVersion', '20.+')}"
+    implementation "com.google.android.gms:play-services-ads:${safeExtGet('googlePlayServicesAdsVersion', '21.+')}"
     implementation 'com.android.support:support-annotations:28.0.0'
     implementation 'androidx.appcompat:appcompat:1.2.0'
     implementation 'com.android.support.constraint:constraint-layout:2.0.4'
diff --git a/node_modules/react-native-admob-native-ads/android/src/main/java/com/ammarahmed/rnadmob/nativeads/RNAdmobNativeView.java b/node_modules/react-native-admob-native-ads/android/src/main/java/com/ammarahmed/rnadmob/nativeads/RNAdmobNativeView.java
index 3a6443b..f6cdb80 100644
--- a/node_modules/react-native-admob-native-ads/android/src/main/java/com/ammarahmed/rnadmob/nativeads/RNAdmobNativeView.java
+++ b/node_modules/react-native-admob-native-ads/android/src/main/java/com/ammarahmed/rnadmob/nativeads/RNAdmobNativeView.java
@@ -126,6 +126,13 @@ public class RNAdmobNativeView extends LinearLayout {
 
             if (ad != null) {
                 nativeAd = ad;
+                nativeAd.setOnPaidEventListener(adValue -> {
+                    WritableMap paidEvent = Arguments.createMap();
+                    paidEvent.putString("currencyCode", adValue.getCurrencyCode());
+                    paidEvent.putDouble("code", adValue.getValueMicros()/ 1000000.0);
+
+                    sendEvent(RNAdmobNativeViewManager.EVENT_AD_FAILED_TO_LOAD, paidEvent);
+                });
                 setNativeAd();
             }
             loadingAd = false;
